tests/src/threading_helpers.c

```diff
@@ -109,50 +109,70 @@ static void mbedtls_test_mutex_usage_error(mbedtls_threading_mutex_t *mutex,
 {
     (void) mutex;
 
-    if (mbedtls_test_info.mutex_usage_error == NULL) {
-        mbedtls_test_info.mutex_usage_error = msg;
-    }
+    mbedtls_test_set_mutex_usage_error(msg);
     mbedtls_fprintf(stdout, "[mutex: %s] ", msg);
     /* Don't mark the test as failed yet. This way, if the test fails later
      * for a functional reason, the test framework will report the message
      * and location for this functional reason. If the test passes,
      * mbedtls_test_mutex_usage_check() will mark it as failed. */
 }
 
+extern mbedtls_threading_mutex_t mbedtls_test_info_mutex;
```

Comment:
I'm not a huge fan of `extern` variables. What about adding a getter also for this address?
However this is just my opinion, so if you prefer to keep it as it is now, then I'm fine ;)